robocop/reports/json_report.py

```diff
@@ -6,14 +9,63 @@ class JsonReport(Report):
     """
     Report name: ``json_report``
 
+    Report that returns list of found issues in JSON format. The output file will be generated
+    in the current working directory with the ``robocop_output.json`` name.
+
+    This report is not included in the default reports. The ``--reports all`` option will not enable this report.
+    You can still enable it using report name directly: ``--reports json_report`` or ``--reports all,json_report``.
+
+    You can configure output directory and report filename::
+
+        robocop --configure json_report:output_dir:C:/json_reports
+        robocop --configure json_report:report_filename:robocop.json
+
+    """
+
+    DEFAULT = False
+
+    def __init__(self):
+        self.name = "json_report"
+        self.description = "Accumulates found issues in JSON format"
+        self.output_dir = None
+        self.report_filename = "robocop_output.json"
+        self.issues = []
+
+    def add_message(self, message: Message):
+        self.issues.append(message.to_json())
+
+    def get_report(self):
+        if self.output_dir is not None:
+            output_path = self.output_dir / self.report_filename
+        else:
+            output_path = Path(self.report_filename)
+        with open(output_path, "w") as fp:
+            json_string = json.dumps(self.issues, indent=4)
+            fp.write(json_string)
+        return f"Generated JSON report in {output_path}"
+
+    def configure(self, name, value):
+        if name == "output_dir":
+            self.output_dir = Path(value)
+            self.output_dir.mkdir(parents=True, exist_ok=True)
+        elif name == "report_filename":
+            self.report_filename = value
+        else:
+            super().configure(name, value)
+
+
+class InternalJsonReport(Report):
+    """
+    Report name: ``internal_json_report``
+
     Report that returns list of found issues in JSON format.
```

Comment:
Maybe we should explain where it "returns" the issues?

Reply:
It is internal report, so it's only for us. But I will add description -> it actually doesn't return anything, it just stores the issues which can be retrieved later.

Reply:
I am thinking if it is not intended to be used by the users, maybe we should mark it in some way? Or just document it here that "Used internally by Robocop, not for external use." or something like that?

Reply:
It is already marked as such, thanks to attributes:

```
    DEFAULT = False
    INTERNAL = True
```

Thanks to INTERNAL attribute it will not be listed in the ``--list-reports``, not present in the docs etc. We have ``json_reports`` for years already and since we added INTERNAL some time ago it was hidden to users. So nothing will change here, we just renamed this internal report and added new one with the same name as old internal one

Reply:
Sounds good to me.